### ISSUE TITLE
Possible Issue with Traefik/MINIO_SERVER_URL/bucket links

### ISSUE THREAD

**Issue by Maypul** [minio/minio#16263]
## Expected Behavior
Deploy docker container, create bucket, generate bucket with correct link.

## Current Behavior
Shared link generated by bucket is local.

## Possible Solution
Possibly use: MINIO_SERVER_URL: https://minio.domain.com but then, I can not login to MinIO instance as it throws an error:
Expected element type <Error> but have <a>

Also when I would use local DNS that redirects my domain to my local device hosting the app, the loading takes long time and throws: Post "https://minio.domain.com/": dial tcp 10.0.0.50:443: i/o timeout as in this case the host is blocking the connection I assume is being made by IP directly.

## Steps to Reproduce (for bugs)
Deploy MinIO behind Traefik Proxy
To get error, use MINIO_SERVER_URL env
To get shared download link generated with local IP, deploy without this env and create bucket, upload file, generate link

## Context
I want to have the app hosted behind Traefik proxy and having the link generated with correct domain form. If I replace the local IP manually with domain name the app is hosted behind it just works. Using the env that possibly sets it right, makes it so that I can not login anymore. Without it, app works but links gets generated with local IP:
Example: http://127.0.0.1:9000/bdbeerbeb/filetest20(...)e319731aaee3d83ef1bda4cc8097090016eb

## Your Environment
Docker compose:
  minio:
    container_name: MinIO
    hostname: minio
    image: quay.io/minio/minio:latest
    restart: unless-stopped
    privileged: false
    read_only: false
    command: server --console-address ":9001" /data
    networks:
      traefik:
    security_opt:
      - no-new-privileges:true
    volumes:
      - $SHR_DOCKER/MinIO/Data:/data
      - $SHR_DOCKER/MinIO/Config:/root/.minio
    secrets:
      - minio_root_account
      - minio_root_password
    environment:
      MINIO_ROOT_USER_FILE: /run/secrets/minio_root_account
      MINIO_ROOT_PASSWORD_FILE: /run/secrets/minio_root_password
      # MINIO_DOMAIN: minio.$DOMAINNAME
      # MINIO_SERVER_URL: https://minio.$DOMAINNAME # <- with this, I can not log in
      MINIO_BROWSER_REDIRECT_URL: https://minio-console.$DOMAINNAME
    labels:
      traefik.enable: true
      traefik.http.routers.minio-rtr.entrypoints: https
      traefik.http.routers.minio-rtr.rule: Host(`minio.$DOMAINNAME`)
      traefik.http.routers.minio-rtr.tls: true
      traefik.http.routers.minio-rtr.middlewares: authelia@file
      traefik.http.routers.minio-rtr.service: minio-svc
      traefik.http.services.minio-svc.loadbalancer.server.port: 9000
      traefik.http.routers.minio-console-rtr.entrypoints: https
      traefik.http.routers.minio-console-rtr.rule: Host(`minio-console.$DOMAINNAME`)
      traefik.http.routers.minio-console-rtr.tls: true
      traefik.http.routers.minio-console-rtr.middlewares: authelia@file
      traefik.http.routers.minio-console-rtr.service: minio-console-svc
      traefik.http.services.minio-console-svc.loadbalancer.server.port: 9001

**Comment by harshavardhana** [comment]
> Possibly use: MINIO_SERVER_URL: https://minio.domain.com but then, I can not login to MinIO instance as it throws an error:
> Expected element type but have

This is because you have a wrong LB configuration it is hitting Console endpoint instead of MinIO API endpoint.

**Comment by harshavardhana** [comment]
> traefik.http.routers.minio-rtr.service: minio-svc

where is this service registered? you are using docker-compose but these values seem to be borrowed from k8s? are you sure you know what you are doing here? 

Going to close this since this is a setup issue.

**Comment by Maypul** [comment]
@harshavardhana thanks, but could you eli5 it to me? What exactly I should have put in these envs? As I have even compared it to some composes I have found on the Internet and I do not see the error.

-traefik.http.routers.minio-rtr.service: minio-svc
It is all docker setup and these are traefik labels. It it for the label below that registers port 9000.

**Comment by harshavardhana** [comment]
> -traefik.http.routers.minio-rtr.service: minio-svc
> It is all docker setup and these are traefik labels. It it for the label below that registers port 9000.

what is actually happening here is quite simply, try first reaching out directly to 

```
curl -k -v https://minio.domain.com/
```

You should see an XML error returned saying "Access Denied" if you don't see it then its a setup issue on your end and traefik is not sending traffic to backend.

**Comment by Maypul** [comment]
@harshavardhana I do see XML with this error at the end when using curl: https://prnt.sc/6eHVFbb2O1AD
This is when running docker image without MINIO_SERVER_URL so I can log in normally.
Now when I do enable this ENV, so MINIO_SERVER_URL: https://minio.domain.com I still get the same Access Denied.
https://prnt.sc/o-fueP4a1C2m but I can not login anymore: https://prnt.sc/Bc3g3Nnudf3c

Also fast addition: It seems to be some kind of DNS issues because after a while the setup was working just fine as expected without any change in config even with the ENV enabled. I will let it run for a while and check later to make sure but... it just worked out of nowhere. So it appears the config above is correct after all.

But also found a workaround for the issue I would have in my future setup where I use local DNS to rewrite my public domain to local IP which would block the server off as it would not allow to connect via IP.

Just add public dns to compose:
    dns:
      - 1.1.1.1
 so the image would get the public IP from the ENV and use that instead of the wrong one.

Anyway thank you for your time!

**Comment by harshavardhana** [comment]
> Now when I do enable this ENV, so MINIO_SERVER_URL: https://minio.domain.com I still get the same Access Denied.
> https://prnt.sc/o-fueP4a1C2m but I can not login anymore: https://prnt.sc/Bc3g3Nnudf3c

This is because your DNS is not resolving to the container, MinIO Console is running on the same container embedded inside MinIO so if you specify MINIO_SERVER_URL it has to be routable from within the docker network through the public domain.

**Comment by Maypul** [comment]
@harshavardhana also noticed one more thing with latest tag images which I think might be worth mentioning. With the latest file download redirection does not work for me, as in the app generates the correct link but it will not download the file until I manually replace the subdomain with the console subdomain. But! Using earlier image with exactly the same setup, the files download (redirects) just fine on the correct subdomain.
As of time of writing, quay.io/minio/minio:latest would not work and quay.io/minio/minio:RELEASE.2022-12-02T19-19-22Z would work. Links in older version do generate longer, no idea if that matters. I do not think it is my setup anymore as I can make it work by using older release.

https://prnt.sc/cIQ_iH0zHE8-
https://prnt.sc/JZxq_kglRrCY

**Comment by harshavardhana** [comment]
> @harshavardhana also noticed one more thing with latest tag images which I think might be worth mentioning. With the latest file download redirection does not work for me, as in the app generates the correct link but it will not download the file until I manually replace the subdomain with the console subdomain. But! Using earlier image with exactly the same setup, the files download (redirects) just fine on the correct subdomain.

what kind of download redirection - can you explain? @Maypul
